File: Lógica de Programação Java e C#/4 - Controle de Fluxo/Java/Exercício de Fixação/EscolheCaminho.java
class EscolheCaminho{
	public static void main(String [] args){
			double valor = Math.random();	
			System.out.println("VALOR = "+valor);
		if(valor < 0.5) {
			System.out.println("Vire à esquerda");
			valor = Math.random();
			System.out.println("Valor = "+valor);
		if (valor < 0.5) {
			System.out.println("Vire a esquerda");
			}else{
				System.out.println("Vire a direita");
			}
		}else{
			System.out.println("Vire a direita");
			valor = Math.random();
			System.out.println("Valor = "+valor);
		if(valor<0.5){
			System.out.println("Vire a esquerda");
			}else{
			System.out.println("Vire a direita");
			}
		}
	}
}

File: Lógica de Programação Java e C#/4 - Controle de Fluxo/C#/Exercícios de Fixação/EscolheCaminho.cs
class EscolheCaminho
{
    static void Main()
    {
        System.Random gerador = new System.Random();
        double valor = gerador.NextDouble();
        System.Console.WriteLine(valor);
        if (valor < 0.5)
        {
            System.Console.WriteLine("Vire a Esquerda");
            valor = gerador.NextDouble();
            System.Console.WriteLine(valor);
            if (valor < 0.5)
            {
                System.Console.WriteLine("Vire a Esquerda");
            }
            else
            {
                System.Console.WriteLine("Vire a Direita");
            }
        }
        else
        {
            System.Console.WriteLine("Vire a direita");
            valor = gerador.NextDouble();
            System.Console.WriteLine(valor);
            if (valor < 0.5)
            {
                System.Console.WriteLine("Vire a Esquerda");
            }
            else
            {
                System.Console.WriteLine("Vire a Direita");
            }
        }
    }
}